invision-app/app/components/home/home.js

```diff
@@ -34,29 +31,46 @@ export class Home extends React.Component {
             movies.sort((movieA, movieB) => movieA.release_date > movieB.release_date ? 1 : -1);
         }
 
-        this.setState({
-            movies
-        });
+        setMovies(movies);
+    }
+    function createUpdateMovie(movie) {
+        if (movies.some((m) => m.id != movie.id)) {
+            movies.push(movie);
+            setMovies(movies)
+        }
+        else {
+            const allmovies = movies.map((m) => {
+                if (m.id === movie.id) {
+                    return movie;
+                }
+                return m;
+            });
+            setMovies(allmovies);
+        }
+        
     }
-    createUpdateMovie=(movie)=> {
-        debugger;
-        console.log(this.state)
-        const { movies }=this.state;
-       movies.push(movie);
-        this.setState({
-            movies
-        });
+    function deleteMovie() {
+        setMovies(movies.filter((m) => m.id !== id));
```

Comment:
All works with state as expected there because **filter** and **map** array functions return new array and do not mutate initial array